torch/_tensor_str.py

```diff
@@ -55,7 +55,7 @@ def set_printoptions(
             PRINT_OPTS.linewidth = 80
         elif profile == "full":
             PRINT_OPTS.precision = 4
-            PRINT_OPTS.threshold = inf
+            PRINT_OPTS.threshold = int(inf)
```

Comment:
This cannot be right:
```
In [2]: from torch._six import inf                                          

In [3]: int(inf)                                                            
---------------------------------------------------------------------------
OverflowError                             Traceback (most recent call last)
<ipython-input-3-8a7cbc05ae2e> in <module>
----> 1 int(inf)

OverflowError: cannot convert float infinity to integer
```

This may not be covered by tests. Just make it a large number, large than the number of elements in a tensor can realistically be. E.g.
```
# Set to an integer value always larger than numel() can be
PRINT_OPTS.threshold = int(1e20)
```

Reply:
Oops, my bad. I've updated the code to use a `Union[int, float]`

Reply:
Just realized that I put `Union[str, float]` rather than `Union[int, float]`